cmd/kubeadm/app/phases/controlplane/manifests_test.go

```diff
@@ -388,6 +388,41 @@ func TestGetAPIServerCommand(t *testing.T) {
 				"--etcd-keyfile=faz",
 			},
 		},
+		{
+                        cfg: &kubeadmapi.MasterConfiguration{
+                                API:               kubeadmapi.API{BindPort: 123, AdvertiseAddress: "2001:db8::1"},
+                                Networking:        kubeadmapi.Networking{ServiceSubnet: "bar"},
+                                CertificatesDir:   testCertsDir,
+                                KubernetesVersion: "v1.9.0-beta.0",
+                        },
+                        expected: []string{
+                                "kube-apiserver",
+                                "--insecure-port=0",
+                                "--admission-control=Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,Priority,ResourceQuota",
+                                "--service-cluster-ip-range=bar",
+                                "--service-account-key-file=" + testCertsDir + "/sa.pub",
+                                "--client-ca-file=" + testCertsDir + "/ca.crt",
+                                "--tls-cert-file=" + testCertsDir + "/apiserver.crt",
+                                "--tls-private-key-file=" + testCertsDir + "/apiserver.key",
+                                "--kubelet-client-certificate=" + testCertsDir + "/apiserver-kubelet-client.crt",
+                                "--kubelet-client-key=" + testCertsDir + "/apiserver-kubelet-client.key",
+                                "--enable-bootstrap-token-auth=true",
+                                "--secure-port=123",
+                                "--allow-privileged=true",
+                                "--kubelet-preferred-address-types=InternalIP,ExternalIP,Hostname",
+                                "--proxy-client-cert-file=/var/lib/certs/front-proxy-client.crt",
+                                "--proxy-client-key-file=/var/lib/certs/front-proxy-client.key",
+                                "--requestheader-username-headers=X-Remote-User",
+                                "--requestheader-group-headers=X-Remote-Group",
+                                "--requestheader-extra-headers-prefix=X-Remote-Extra-",
+                                "--requestheader-client-ca-file=" + testCertsDir + "/front-proxy-ca.crt",
+                                "--requestheader-allowed-names=front-proxy-client",
+                                "--authorization-mode=Node,RBAC",
+                                "--advertise-address=1.2.3.4",
+                                "--etcd-servers=http://127.0.0.1:2379",
+                        },
+                },
```

Comment:
then when you have fixed the above comments, copy the newly added code and make a nearly similar test case for `CloudProvider: aws`